### PR TITLE
[Popups]: added destroy() function in all popups by default

### LINKED ISSUE #10149: Review usages of Global.openPopup
## Description

`Global.openPopup` dynamically creates popups which needs to be deleted by the caller using `popup.destroy()`. Otherwise the popup will reside in the memory to the end of the app execution. It's easy to forget about deletion, probably it would be beneficial to remove the popup by default on close. The exception are popups intentionally kept in memory e.g. emoji popup.

### PR DESCRIPTION
**Pull request by alexandraB99**
* Except Emoji and Stickers popups

Also switched StatusModal to derive from StatusDialog instead of Popup type

Closes #10149

### What does the PR do
Popups, added destroy() function in all popups by default

### Affected areas
Popups

### CODE REVIEW

File: ui/StatusQ/src/StatusQ/Controls/StatusComboBox.qml [hunk 1]
```diff
@@ -157,6 +157,7 @@ Item {
                     model: comboBox.popup.visible ? comboBox.delegateModel : null
                     currentIndex: comboBox.highlightedIndex
                 }
+                onClosed: { destroy(); }
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Controls/StatusComboBox.qml:160]: Not sure it's desired here; it's part of the component, there's no need to destroy this if we hadn't created it explicitly ourselves

File: ui/app/AppLayouts/Browser/popups/BrowserWalletMenu.qml [hunk 2]
```diff
@@ -212,4 +212,7 @@ Popup {
             }
         }
     }
+    onClosed: {
+        popup.destroy();
```
Comment (caybro) [inline ui/app/AppLayouts/Browser/popups/BrowserWalletMenu.qml:216]: Same here

File: ui/app/mainui/Popups.qml [hunk 3]
```diff
@@ -338,7 +327,7 @@ QtObject {
                 profileStore: rootStore.profileSectionStore.profileStore
                 contactsStore: rootStore.profileSectionStore.contactsStore
                 communitiesModel: rootStore.profileSectionStore.communitiesList
-
+                destroyOnClose: false
```
Comment (caybro) [inline ui/app/mainui/Popups.qml:330]: I wonder... why not here then?

Comment (alexandraB99) [reply ui/app/mainui/Popups.qml:330]: in this case needs to do stuff before closing so the destroy is called inside onClose:
```

                onClosed: {
                    if (profilePopup.parentPopup) {
                        profilePopup.parentPopup.close()
                    }
                    destroy()
                }
```

File: ui/app/AppLayouts/Browser/popups/BrowserWalletMenu.qml (hunk 2 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Browser/popups/BrowserWalletMenu.qml:216]: hmm this is a `Popup` type, wdym?

Comment (caybro) [reply ui/app/AppLayouts/Browser/popups/BrowserWalletMenu.qml:216]: Yeah but did we create it ourselves explicitly?

Comment (alexandraB99) [reply ui/app/AppLayouts/Browser/popups/BrowserWalletMenu.qml:216]: yesh, it's instantiated in BrowserLayout.qml line 427

**Review by caybro — APPROVED** [review]

**Review by alexjba — APPROVED** [review]

**Comment by alexjba** [comment]
Do we need QA here? I'm thinking if we could miss cases where a sequence like `open-close-open` would fail on the second `open` because the popup was destroyed.

**Comment by borismelnik** [comment]
@alexandraB99 what will happens if someone create the popup not it `Component`?

**Comment by borismelnik** [comment]
Maybe better do something like this in `Global.openPopup`:
```
popup.closed.connect(popup.destroy)
```?

**Comment by caybro** [comment]
> Maybe better do something like this in `Global.openPopup`:
> 
> ```
> popup.closed.connect(popup.destroy)
> ```
> ?

We don't want to do it everywhere unconditionally

**Comment by borismelnik** [comment]
> > Maybe better do something like this in `Global.openPopup`:
> > ```
> > popup.closed.connect(popup.destroy)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ?
> 
> We don't want to do it everywhere unconditionally

Do you mean its impossible to do it centralized?

**Review by Cuteivist — APPROVED** [review]

**Comment by alexandraB99** [comment]
> Do we need QA here? I'm thinking if we could miss cases where a sequence like `open-close-open` would fail on the second `open` because the popup was destroyed.

@anastasiyaig please consider this when testing

**Review by anastasiyaig — APPROVED** [review]

**Review by micieslak — APPROVED** [review]

File: ui/StatusQ/src/StatusQ/Popups/Dialog/StatusDialog.qml [hunk 4]
```diff
@@ -11,6 +11,16 @@ Dialog {
...
+    /*!
+       \qmlproperty destroyOnClose
+        This property decides whether the popup component should be destroyed when closed. Default value is true.
+    */
+    property bool destroyOnClose: true
+    /*!
+       \qmlproperty color backgroundColor
+        This property decides the modal background color
+    */
+    property string backgroundColor: Theme.palette.statusModal.backgroundColor
```
Comment (osmaczko) [inline ui/StatusQ/src/StatusQ/Popups/Dialog/StatusDialog.qml:23]: https://github.com/status-im/StatusQ/pull/809#pullrequestreview-1054132211

File: ui/StatusQ/src/StatusQ/Popups/Dialog/StatusDialog.qml [hunk 5]
```diff
@@ -101,4 +113,9 @@ Dialog {
             }
         }
     }
+
+    onClosed: {
+        if (root.destroyOnClose)
+            root.destroy();
+    }
```
Comment (osmaczko) [inline ui/StatusQ/src/StatusQ/Popups/Dialog/StatusDialog.qml:120]: It does not belong here, it is not the responsibility of the Dialog.  The lifetime of objects should be managed by the entity that creates them.

**Review by osmaczko — COMMENTED** [review]
@alexandraB99 I noticed that I'm listed as a co-author of this commit and I'm unsure why. Could you please help me understand how this happened? I don't recall contributing directly to these changes.

**Pull request merged**